Ghidra/Features/Base/src/main/java/ghidra/plugins/fsbrowser/FSBComponentProvider.java

```diff
@@ -192,8 +194,20 @@ private void renderFile(FSBFileNode node, boolean selected) {
 					overlays.add(FSBIcons.MISSING_PASSWORD_OVERLAY_ICON);
 				}
 
-				Icon icon = fsbIcons.getIcon(filename, overlays);
-				setIcon(icon);
+				GFile gf = node.getGFile();
+				GFileSystem fs = gf.getFilesystem();
+				// TODO is it OK to pass a null monitor?
+				FileAttributes fa = fs.getFileAttributes(gf, null/*monitor*/);
+
+				String ext = fa.get("typeExt", String.class, "");
```

Comment:
I would add a new FileAttributeType, maybe something like:

```FILENAME_EXT_OVERRIDE("Extension override", MISC_INFO, String.class)```

right after ```COMMENT_ATTR```, and then in FSBFileNode.updateFileProps(), query for it in a similar manner as the other values, and save it for later use.

Then, instead of adding a new fsbIcons.getIconByExt() method, seems like we could just smush the existing filename together with the extension override and then query the icon in the normal way.

A bit of a hack, but pretty clear and easy to understand and easy to 'fix' later on if we run into a requirement  where we need more feature-rich file typing.

Reply:
additionally, there should be some logic that ensures the extension override value is formatted (or not formatted) with a leading dot '.', so that when its used to concat with the filename we get a good value, regardless of what the filesystem impl throws at us.